internal/venice-common/src/main/java/com/linkedin/venice/writer/VeniceWriter.java

```diff
@@ -1081,7 +1111,13 @@ private Future<RecordMetadata> sendMessage(
       }
 
       try {
-        return producer.sendMessage(topicName, key, kafkaValue, partition, messageCallback);
+        ProducerRecord<KafkaKey, KafkaMessageEnvelope> kafkaRecord = new ProducerRecord<>(
+            topicName,
+            partition,
+            key,
+            kafkaValue,
+            key.isControlMessage() ? protocolSchemaHeaders : emptyHeaders);
```

Comment:
Will this introduce a lot of extra usage of Kafka?
So the schema header will be sent for every Control Message, which includes SOS and EOS. Because of the frequent segmentation logic, there will be a new segment every 5 min and the Control Messages are not compactable.. 

Shall we only send the schema header in the first Control message per producer guid per partition?

Reply:
I suppose we could send it in the first one only, since the later ones are in theory useless.

Reply:
I've made this change in the latest commit.